omegaconf/dictconfig.py

```diff
@@ -372,8 +372,21 @@ def __getitem__(self, key: DictKeyType) -> Any:
         except Exception as e:
             self._format_and_raise(key=key, value=None, cause=e)
 
+    def __delattr__(self, key: str) -> None:
+        """
+        Allow deleting dictionary values as attributes
+        :param key:
+        :return:
+        """
+        if key == "__name__":
+            raise AttributeError()
+        self._delkey(key)
```

Comment:
Deleting from an object raises an AttributeError.
When deleting an attribute from a DictConfig backed by a Structured Config I think we should raise ConfigAttributeError.

Reply:
Did I miss your response to this?
I am not seeing a reply, and I am not seeing this addressed either.

Reply:
FYI @jieru-hu I think that in this case the right way to check if a DictConfig is backed by a Structured Config would be to evaluate `is_structured_config(self._metadata.object_type)`.

Reply:
Sorry I missed this one! I've updated the PR to address this comment.

Reply:
Oh, please disregard my previous comment, what you've got now looks good (raising ConfigAttributeError for both structured and non-structured)